### DISCUSSION TITLE
Security of Apple IRK

### DISCUSSION THREAD

**Discussion by formbuddy** [ESPresense/ESPresense#814] [General]
Thanks for building and maintaining this awesome project. I recently started using ESPresense and have enrolled my iPhone using IRK.

While I understand that the intent behind IRK support is to resolve conflicts between different devices, but does IRK also have a side effect of increased security? Is the IRK visible only to paired devices or can any Bluetooth scanner access the IRK? I am evaluating to see if this could be a reasonable solution for opening smart locks. I do understand that any paired device could potentially have access to the IRK.

**Comment by raptile** [comment]
only paired devices are able to get the IRK.

A resolvable private address is, true to its name, resolvable as intended listeners have a pre-shared key by which they can figure out the new address every time it changes. The pre-shared key is the Identity Resolving Key (IRK) and is used both to generate and to resolve the random address.

The random address is basically just used by the peer to be able to resolve the actual address of the Bluetooth LE device, which is still either the public or the random static address. The IRK allows the peer to translate the random private address into the device’s real Bluetooth LE address.
